infra/graph_generators_test.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Wed Mar 20 17:17:54 2019
+
+@author: Zargham
+"""
+import networkx as nx
+import unittest
+from graph_generators import lineGraphGen, starGraphGen, treeGraphGen
+
+class GraphGenerators_LineGraph(unittest.TestCase):
+    def test_lineGraphGen_sanity(self):
+        g = lineGraphGen(4)
+        self.assertEqual(list(g.edges()), [(0, 1), (1, 2), (2, 3)])
+
+    def test_lineGraphGen_bidir(self):
+        g = lineGraphGen(4, bidir=True)
+        # (node, neighbor)
+        self.assertEqual(list(g.edges()), [(0, 1), (1, 0), (1, 2), (2, 1), (2, 3), (3, 2)])
+
+class GraphGenerators_StarGraph(unittest.TestCase):
+
+# how to validate that 'kind' is either 'source', 'sink', or 'bidir' ?? 
```

Comment:
we should have the fn raise a ValueError if the kind isnt an acceptable value, then test that behavior here.

Reply:
added error handling (first try)

Reply:
Added error handling across the module.